pkg/authentication/authentication.go

```diff
@@ -281,9 +313,20 @@ func (a *Handler) handleSuccessfulAuth(w http.ResponseWriter, r *http.Request, g
 
 	if r.Header.Get("Accept") == "application/json" {
 		accountProviders, _ := dbUser.GetAccountProviders()
+
+		// Extract primary email and avatar from account providers
+		primaryEmail := ""
+		primaryAvatar := ""
+		if len(accountProviders) > 0 {
+			primaryEmail = accountProviders[0].Email
+			primaryAvatar = accountProviders[0].AvatarURL
+		}
```

Comment:
This is repeated multiple times, we can put it in a function like `getPrimaryEmail(accountProviders)` and `getPrimaryAvatar(accountProviders)`